lib/krane/psych_k8s_patch.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'psych'
+
+# PsychK8sPatch applies patching to Psych.dump/dump_stream with an alternative string serializer that is more compatible
+# with Kubernetes (and other Go-based tooling). Related issue: https://github.com/Shopify/krane/issues/740
+module PsychK8sPatch
+  # Activate will apply the patch after creating backup aliases for the original methods.
+  def self.activate
+    class << Psych
```

Comment:
See https://github.com/Shopify/krane/pull/742 for an alternative implementation

Reply:
Interesting! One second let me put together some thoughts

Reply:
Okay, so lots of thoughts on this.

First of all, this solution definitely works. I don't think there's anything particularly wrong with it at all, other than it _feels_ tough to stomach since it's a monkey-patch.

I think what Tim is getting at with his solution is scoping the monkey-patch locally with a refinement. I think a refinement could also work here, but since there's only one place where we're truly dumping out to YAML I think we can be even more explicit by just defining the behavior we want.

Fundamentally it seems like (and please correct my if I'm wrong here) we're trying to change the behavior so that it uses double quotes when we're visiting a string that represents some scientific notation. I think a more clear way to achieve that would be to explicitly define our own visitor and then use it. As in:

```ruby
module Psych
  module Visitors
    class ScientificYAMLTree < YAMLTree
      def visit_String(string)
        return super unless string.match?(/\A[+-]?\d+(?:\.\d+)?[eE][+-]?\d+\z/)
        @emitter.scalar(string, nil, nil, true, true, Nodes::Scalar::DOUBLE_QUOTED)
      end
    end
  end

  def self.dump_scientific(object)
    visitor = Visitors::ScientificYAMLTree.create
    visitor << object
    visitor.tree.yaml
  end
end
```

Then with that in place, calling something like `YAML.dump_scientific('123e4')` will yield `"--- \"123e4\"\n"`. I like this a little better personally because it touches slightly less internals in `Psych`, it's a locally scoped change, and it's explicit about the behavior that we want changed.